File: CPP-Source/Day17.cpp
//DAY 17 OF HACKERRANK 30 DAY CHALLENGE: MORE EXCEPTIONS (C++)

#include <iostream>
#include <exception>
#include <stdexcept>

using namespace std;

class powerexception: public exception {
    const char* what() const throw() {
        return "n and p should be non-negative";
    }
};

class Calculator {
    public:
        Calculator() {
            //Do nothing;
        }
        
        int power (int n, int p) {
            if ((n >= 0) && (p >= 0)) {
                int output = 1;
                for (int x = 0; x < p; x++) {
                    output = output * n;
                }
                return output;
            }
            else {
                powerexception px;
                throw px;
                return -1;
            }
        }
};

int main()
{
    Calculator myCalculator=Calculator();
    int T,n,p;
    cin>>T;
    while(T-->0){
      if(scanf("%d %d",&n,&p)==2){
         try{
               int ans=myCalculator.power(n,p);
               cout<<ans<<endl; 
         }
         catch(exception& e){
             cout<<e.what()<<endl;
         }
      }
    }
    
}

File: Java-Source/Day17.java
//DAY 17 OF HACKERRANK 30 DAY CHALLENGE: MORE EXCEPTIONS (JAVA)

import java.util.*;

class Calculator {
    int n;
    int p;

    Calculator() {
        n = 0;
        p = 0;
    }

    public int power (int n, int p) throws Exception {
        int output = 1;
        if ((p >= 0) && (n >= 0)) {
            for (int x  = 0; x < p; x++) {
                output = output * n;
            }

            return output;
        }
        else {
            throw new Exception("n and p should be non-negative");
        }
    }
}

public class Day17 {

    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        while (t-- > 0) {

            int n = in.nextInt();
            int p = in.nextInt();
            Calculator myCalculator = new Calculator();
            try {
                int ans = myCalculator.power(n, p);
                System.out.println(ans);
            }
            catch (Exception e) {
                System.out.println(e.getMessage());
            }
        }
        in.close();
    }
}
